dagshub/streaming/filesystem.py

```diff
@@ -345,28 +345,23 @@ def listdir(self, path='.'):
     def scandir(self, path='.'):
         path = Path(path)
         relative_path = self._relative_path(path)
-        if relative_path:
-            if self._passthrough_path(relative_path):
-                with self._open_fd(relative_path) as fd:
-                    return self.__scandir(fd)
-            else:
-                local_filenames = set()
-                try:
-                    with self._open_fd(relative_path) as fd:
-                        for direntry in self.__scandir(fd):
-                            local_filenames.add(direntry.name)
-                            yield direntry
-                    if relative_path == Path():
-                        if SPECIAL_FILE.name not in local_filenames:
-                            yield dagshub_DirEntry(self, path / SPECIAL_FILE, is_directory=False)
-                except FileNotFoundError:
-                    pass
-                resp = self._api_listdir(relative_path)
-                if resp.ok:
-                    for f in resp.json():
-                        name = Path(f['path']).name
-                        if name not in local_filenames:
-                            yield dagshub_DirEntry(self, path / name, f['type'] == 'dir')
+        if relative_path and not self._passthrough_path(relative_path):
```

Comment:
Maybe `passthrough_path` should check if the relative path starts with `..` as well.

Reply:
relative_path is None if it's not actually inside of the dir. It doesn't perform any fs walking or smth like that to figure out the relative path outside. Only inside